src/mysql_helpers.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Helper class to manage the MySQL Router lifecycle."""
+
+import logging
+import socket
+
+import mysql.connector
+from tenacity import retry, stop_after_delay, wait_fixed
+
+logger = logging.getLogger(__name__)
+
+
+class Error(Exception):
+    """Base class for exceptions in this module."""
+
+    def __repr__(self):
+        """String representation of the Error class."""
+        return "<{}.{} {}>".format(type(self).__module__, type(self).__name__, self.args)
+
+    @property
+    def name(self):
+        """Return a string representation of the model plus class."""
+        return "<{}.{}>".format(type(self).__module__, type(self).__name__)
+
+    @property
+    def message(self):
+        """Return the message passed as an argument."""
+        return self.args[0]
+
+
+class MySQLRouterCreateUserWithDatabasePrivilegesError(Error):
+    """Exception raised when there is an issue creating a database scoped user."""
+
+
+class MySQLRouterPortsNotOpenError(Error):
+    """Exception raised when mysqlrouter is not bootstrapped and started."""
+
+
+class MySQL:
```

Comment:
Perhaps this could be renamed `MySQLRouter` to better define it from the `MySQL` class in the other charm?

Reply:
renamed in b22efe12e0d56b70d439b490de176040b0e1bd8f